src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableValueStore.kt

```diff
@@ -226,13 +225,21 @@ class VariableValueStore(
   }
 
   // Get the VariableValue for the injected variable within a section variable
-  fun fetchSectionValueVariable(
-      variableId: VariableId
-  ): List<VariableValue<VariableValueId, SectionValueVariable>> {
-    val conditions = listOf(VARIABLE_SECTION_VALUES.USED_VARIABLE_ID.eq(variableId))
+  private fun fetchSectionValuesUsingVariable(variableId: VariableId): List<ExistingSectionValue> {
+    val variableValues2 = VARIABLE_VALUES.`as`("variable_values_2")
+    val conditions =
+        listOf(
+            VARIABLE_SECTION_VALUES.USED_VARIABLE_ID.eq(variableId),
+            // Don't return values that were later superseded by other values
```

Comment:
🙏

Reply:
Can you add a test case to verify that this works as intended and that we don't erroneously upgrade variables from previous versions of a section?